src/usng.ts

```diff
@@ -28,1352 +28,1380 @@
 
 /*global define,Math,parseInt,parseFloat*/
 
+type LL = {
+  lat: number,
+  lon: number
+};
+
+type BBox = {
+  north: number,
+  south: number,
+  east: number,
+  west: number
+};
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/atanh
+const atanh = Math["atanh"] || function(x) {
+  return Math.log((1+x)/(1-x)) / 2;
+};
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON
+const epsilon = typeof Number["EPSILON"] === "undefined" ? Math.pow(2, -52) : Number["EPSILON"];
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/hypot
+const hypot = Math["hypot"] || function (x, y) {
+  // https://bugzilla.mozilla.org/show_bug.cgi?id=896264#c28
+  var max = 0;
+  var s = 0;
+  for (var i = 0; i < arguments.length; i += 1) {
+    var arg = Math.abs(Number(arguments[i]));
+    if (arg > max) {
+      s *= (max / arg) * (max / arg);
+      max = arg;
+    }
+    s += arg === 0 && max === 0 ? 0 : (arg / max) * (arg / max);
```

Comment:
✏️ This line is rather obfuscated. I suppose if I had a better internal picture of the order of precedence here I could ascertain what `s` would equal at the end, but right now I'm just guessing that the increment to `s + arg` will be tossed out and `s` will end up equaling either the left or right output.

Would read more clearly (assuming that interpretation is correct) as
`s = ((s + arg) === 0 && max === 0) ? 0 : (arg / max) * (arg / max)`

Unless that increment occurs later...is it
`s += ((arg === 0 && max === 0) ? 0 : (arg / max) * (arg / max))`

Reply:
This code is part of ES5 polyfill from the link atop the function that contains this line. I am hesistant to change any code in here for that reason.

Reply:
And people wonder why so many engineers despise JavaScript.

This polyfill was written by, presumably, stewards of the language.

Reply:
Agreed and correct; this is from the JavaScript documentation from Mozilla.